html2asketch/nodeToSketchLayers.js

```diff
@@ -228,6 +228,13 @@ export default function nodeToSketchLayers(node, options) {
     return layers;
   }
 
+  // Traverse the shadow DOM if present
+  if (node.shadowRoot) {
+    Array.from(node.shadowRoot.children)
+      .map(nodeToSketchLayers)
```

Comment:
we are going recursive here which wasn't the case before 🤔 I'm not sure if we should do that, `nodeToSketchLayers` was always about changing single node into layers and ignoring its children. Higher level solutions, such as `nodeTreeToSketchGroup`, should do the traversing. Will your `nodeTreeToSketchGroup` implementation work if we don't include this change in `nodeToSketchLayers`?

Reply:
It looks like this recursiveness here produces invalid results for `nodeTreeToSketchGroup`:

<img width="372" alt="screen shot 2018-05-28 at 12 04 59" src="https://user-images.githubusercontent.com/985504/40610001-de084616-6270-11e8-9e6a-c7745d3d6743.png">

Lets remove it from here 👍

Reply:
So technically `nodeToSketchLayers` layers is still a single node of you imagine that each custom element node with shadow DOM attached is representative of a single element (like `HTMLVideoElement` is a closed shadow root with buttons etc. 

Not sure why you're getting different results visually than I am, that's working perfectly on my end. I'll take a look.

Reply:
`nodeToSketchTreeGroup` works fine, will not including this change affect the ability to turn shadow DOM elements into symbols? Or am I misunderstanding what the difference between the tools are?

Reply:
@kdzwinel Also, the clean install didn't change anything in terms of the e2e tests. Maybe I'll try setting this up on my personal computer.

Reply:
> Not sure why you're getting different results visually than I am,

I was running `nodeTreeToSketchGroup ` to produce that screenshot. The issue is that both:
- nodeToSketchLayers (https://github.com/brainly/html-sketchapp/pull/103/files#diff-fa6ecde3539d30f898e6d090901f2262R23) and
- nodeToSketchTreeGroup (https://github.com/brainly/html-sketchapp/pull/103/files#diff-cb9e62615b812bac77eac76d375e2308R232) are doing node traversing.
This causes problems when running `nodeTreeToSketchGroup ` (which uses `nodeToSketchLayers`) because some of the nodes get traversed, and converted, twice.

> will not including this change affect the ability to turn shadow DOM elements into symbols?

hm, good point. You'd have to use `nodeTreeToSketchGroup` (instead of `nodeToSketchLayers`) to get a sketch group from a node and then add it to the symbol via `addLayer`. Which, now that I think about it, makes sense. If someone wants to deal with DOM traversing themselves they can use `nodeToSketchLayers`, but in any other case they should use `nodeTreeToSketchGroup`.

> Maybe I'll try setting this up on my personal computer.

hm, please do check from your personal computer. Maybe you are getting different Chromium version at work? Not sure where these differences can be coming from :/

Reply:
@kdzwinel The latest commit was done with my personal computer, the output was different than my work computer, but still not what the Travis build expected. Any ideas here?